extensions/tokenizers/src/main/java/ai/djl/huggingface/tokenizers/Encoding.java

```diff
@@ -55,23 +56,85 @@ protected Encoding(
         this.overflowing = overflowing;
     }
 
+    /**
+     * Returns the {@link NDList} representation of the encodings.
+     *
+     * @param encodings the {@code Encoding} batch
+     * @param manager the {@link NDManager} to create the NDList
+     * @param withTokenType true to include the token type id
+     * @param int32 true to use int32 datatype
+     * @return the {@link NDList}
+     */
+    public static NDList toNDList(
+            Encoding[] encodings, NDManager manager, boolean withTokenType, boolean int32) {
```

Comment:
could we leverage the instance method toNdList here? It seems like we should be able to loop over all the encondings and call encoding.toNdList(...)?

Reply:
They are different, the static one is batched (`Encoding[]`)

Reply:
yes, but per encoding they are the same. Why can't we do something like this?

```
public static NDList toNDList(Encoding[] encodings, NDManager manager, boolean withTokenType, boolean int32) {
    NDList list = new NDList();
    for (int i = 0; i < encodings.length; i++) {
        NDList encoding = encodings[i].toNDList(manager, withTokenType, int32);
        list.addAll(encoding);
    }
    return list;
}
```

Reply:
This is an optimization for performance (was trying to compete with TEI). It reduced number of memory copies to create a batched NDArray. This way it only invoke JNI 3 times, while the loop one will copy N times and then use stack to batchify them.

I actually not sure if this is necessary.